qemu/tests/block_kill_reconnect_with_remote_storage.py

```diff
@@ -0,0 +1,157 @@
+import logging
+import time
+
+from provider.nbd_image_export import QemuNBDExportImage
+from provider.blockdev_base import BlockdevBaseTest
+from virttest import error_context
+
+from avocado.utils import process
+
+
+class BlockReconnectTest(BlockdevBaseTest):
+    """
+    Block mirror with firewall test
+    """
+
+    def __init__(self, test, params, env):
+        self.net_down = False
+
+        super(BlockReconnectTest, self).__init__(test, params, env)
+
+    def prepare_test(self):
+        try:
+            self.create_local_image()
+            self.export_local_image_with_nbd()
+            super(BlockReconnectTest, self).prepare_test()
+        except Exception:
+            self.clean_images(self.net_down)
+            raise
+
+    def create_local_image(self):
+        image_params = self.params.object_params(
+            self.params.get("local_image_tag"))
+        local_image = self.source_disk_define_by_params(
+            image_params, self.params.get("local_image_tag"))
+        local_image.create(image_params)
+        self.trash.append(local_image)
+
+    def export_local_image_with_nbd(self):
```

Comment:
here we export the local image with local nbd server(localhost or hostname), I didn't see you define nbd_server_data1, so you may use the same nbd_server with your nbd storage
e.g. nbd server: 1.2.3.4
test server: a.b.c.d
here you should use a.b.c.d instead of 1.2.3.4

Reply:
Thanks for your comment, network-storage tests all use local nbd server, it define in /tmp/nbd.cfg, nbd_server = hostname, unless specify other server, it should use the default one. Hence, I may also delete option nbd_server_tls_creds_data.

Reply:
You mean your testing machine and nbd server machine are the same one? I suggest that we decouple them, for we cannot assume that they are the same one, even though we can run our tests on the nbd server, i.e. the code should not depend on it, it should work fine when nbd server and testing machine are not the same one.

Reply:
Support para like nbd_server_data1 = xxx, default is localhost.